### PR TITLE
Command line argument formatting

### PR DESCRIPTION
**Pull request by sb2nov**

### CODE REVIEW

File: dataduct/s3/utils.py [hunk 1]
```diff
@@ -158,3 +158,35 @@ def delete_dir_from_s3(s3_path):
...
+
+    Args:
+        s3_old_path(S3Path): Output path of the file to be uploaded
+        s3_new_path(S3Path): Output path of the file to be uploaded
+        raise_when_no_exist(bool, optional): Raise error if file not found
+
+    Raises:
+        ETLInputError: If s3_old_path does not exist
+    """
+    assert isinstance(s3_old_path, S3Path), 'old path should be of type S3Path'
```
Comment (tpcstld) [inline dataduct/s3/utils.py]: Change to raise Exceptions? I actually prefer assertions though ):

File: dataduct/s3/utils.py [hunk 2]
```diff
@@ -158,3 +158,35 @@ def delete_dir_from_s3(s3_path):
...
+    assert s3_old_path.is_directory, 'S3 old path must be directory'
+    assert isinstance(s3_new_path, S3Path), 'new path should be of type S3Path'
+    assert s3_new_path.is_directory, 'S3 new path must be directory'
+
+    bucket = get_s3_bucket(s3_old_path.bucket)
+    prefix = s3_old_path.key
+
+    # Enforce this to be a folder's prefix
+    if not prefix.endswith('/'):
```
Comment (tpcstld) [inline dataduct/s3/utils.py]: One-line this

File: dataduct/s3/utils.py [hunk 3]
```diff
@@ -158,3 +158,35 @@ def delete_dir_from_s3(s3_path):
...
+        prefix += '/'
+    keys = bucket.get_all_keys(prefix=s3_old_path.key)
+    for key in keys:
+        if key:
```
Comment (tpcstld) [inline dataduct/s3/utils.py]: `if key is not None`?

File: dataduct/s3/utils.py [hunk 4]
```diff
@@ -158,3 +158,35 @@ def delete_dir_from_s3(s3_path):
...
+            key.copy(s3_new_path.bucket,
+                     os.path.join(s3_new_path.key, os.path.basename(key.key)))
+
+        if raise_when_no_exist and not key:
```
Comment (tpcstld) [inline dataduct/s3/utils.py]:
```
else:
    if raise_when_no_exist
```

File: dataduct/utils/cli.py [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+from argparse import ArgumentParser
+from argparse import RawTextHelpFormatter
+import argparse
+
+
+def config_singleton_setup(args):
+    """Setup the config singleton based on the mode in args
+    """
+    mode = args.mode if hasattr(args, 'mode') else None
+    import logging
```
Comment (tpcstld) [inline dataduct/utils/cli.py]: Comment about why you import here instead of at the top

File: dataduct/utils/cli.py [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    logger = logging.getLogger(__name__)
+
+    from dataduct.config import Config
+    from dataduct.config import logger_configuration
+
+
```
Comment (tpcstld) [inline dataduct/utils/cli.py]: remove one new line from the two

File: dataduct/utils/cli.py [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+    # Setup up logging for package
+    logger_configuration()
+
+    if mode is not None:
+        logger.warning('Running in %s mode', config.mode)
+    return config
+
+
+# Override help action for better help output
```
Comment (tpcstld) [inline dataduct/utils/cli.py]: Move this into the docstring as a note

File: dataduct/utils/cli.py [hunk 8]
```diff
@@ -0,0 +1,146 @@
...
+class DataductHelpAction(argparse._HelpAction):
+    """HelpAction class used to render a custom help message
+    """
+    def __call__(self, parser, namespace, values, option_string=None):
+        parser.print_help()
+        print ''
```
Comment (tpcstld) [inline dataduct/utils/cli.py]: `# Empty line`

File: dataduct/utils/cli.py [hunk 9]
```diff
@@ -0,0 +1,146 @@
...
+        # retrieve subparsers from parser
```
Comment (tpcstld) [inline dataduct/utils/cli.py]: Capitalize the first letter of these comments

File: dataduct/utils/cli.py [hunk 10]
```diff
@@ -0,0 +1,146 @@
...
+pipeline_run_options = ArgumentParser(
+    description='Specify actions related to running the pipelines',
+    add_help=False
+)
+pipeline_run_options.add_argument(
+    '-f',
+    '--force',
+    action='store_true',
+    default=False,
+    help='Indicates that if this pipeline exists, it will be destroyed',
```
Comment (tpcstld) [inline dataduct/utils/cli.py]: I would suggest "Destroy previous versions of this pipeline, if they exist"

File: dataduct/utils/cli.py [hunk 11]
```diff
@@ -0,0 +1,146 @@
...
+mode_parser.add_argument(
+    '-m',
+    '--mode',
+    default=None,
+    help=mode_help
+)
+
+# Options parser shared actions all pipeline run options
+pipeline_run_options = ArgumentParser(
+    description='Specify actions related to running the pipelines',
```
Comment (tpcstld) [inline dataduct/utils/cli.py]: "running the pipeline" or "running pipelines"

File: dataduct/utils/cli.py [hunk 12]
```diff
@@ -0,0 +1,146 @@
...
+    '-b',
+    '--backfill',
+    action='store_true',
+    default=False,
+    help='Indicates that the timedelta supplied is for a backfill',
+)
+pipeline_run_options.add_argument(
+    '--frequency',
+    default=None,
+    help='Frequency override to the pipeline',
```
Comment (tpcstld) [inline dataduct/utils/cli.py]: for the pipeline

File: dataduct/utils/cli.py [hunk 13]
```diff
@@ -0,0 +1,146 @@
...
+    add_help=False,
+)
+table_definition_parser.add_argument(
+    'table_definitions',
+    nargs='+',
+    help=table_definition_help,
+)
+
+# Filepath input parser
+filepath_help = 'filepath input for storing output of actions'
```
Comment (tpcstld) [inline dataduct/utils/cli.py]: Capital first letter

**Comment by tpcstld** [comment]
Everything else LGTM

File: dataduct/s3/utils.py (hunk 3 above)
Comment (sb2nov) [inline dataduct/s3/utils.py]: Actually i'm not sure what boto returns so I would keep it as is before I find out more.

**Pull request merged**